include/internal/catch_test_case_registry_impl.h

```diff
@@ -23,6 +23,8 @@ namespace Catch {
     struct IConfig;
 
     std::vector<TestCase> sortTests( IConfig const& config, std::vector<TestCase> const& unsortedTestCases );
+
+    bool isThrowSafe( TestCase const& testCase, IConfig const& config );
```

Comment:
I have two nitpicks with this function.

1) I don't like the fact that it is declared in two different headers. Does it need to be declared in both?
2) I don't like the name (but I currently don't have a better one either).

Reply:
Hmm, I suppose it doesn't, but then wouldn't that also go for the other 3 functions declared in catch_test_case_registry.h? I added it just to reuse this part of the condition and figured it should have the same availability as `matchTest`.

Without changes to `#include`s in other files, my build succeeds with the declarations in a single location as follows:
1. catch_test_case_registry.h:
  Must have `isThrowSafe`, `filterTests`, `getAllTestCasesSorted`, may have `matchTest`;
2. catch_test_case_registry_impl.h:
  May have `matchTest`, doesn't need the other three.

So there are some locations that can only see the first header. What would be most appropriate? I'll wait before pushing another commit to resolve this.

Reply:
Hmmm, let's leave it be for now. I'll be doing a significant rewrite of those parts of the code for v3 anyway.

Reply:
Alright. As for the name `isThrowSafe`, the only other idea that came to mind was something like `noIllegalThrows`, as `!(testCase.throws() && !config.allowThrows())` is a less readable equivalent of the condition. A slight downside is that it has a double negation, but that may be permissible since `!isThrowSafe` is not used.

I'm leaving it as is so you may decide whether to change it.